src/ros_filter.cpp

```diff
@@ -2178,30 +2178,29 @@ void RosFilter<T>::periodicUpdate()
       if (filtered_position->header.frame_id == odom_frame_id_) {
         world_transform_broadcaster_->sendTransform(world_base_link_trans_msg_);
       } else if (filtered_position->header.frame_id == map_frame_id_) {
-        try {
-          tf2::Transform world_base_link_trans;
-          tf2::fromMsg(
-            world_base_link_trans_msg_.transform,
-            world_base_link_trans);
-
-          tf2::Transform base_link_odom_trans;
-          tf2::fromMsg(
-            tf_buffer_
-            ->lookupTransform(
-              base_link_frame_id_,
-              odom_frame_id_,
-              tf2::TimePointZero)
-            .transform,
-            base_link_odom_trans);
-
+        tf2::Transform world_base_link_trans;
+        tf2::fromMsg(
+          world_base_link_trans_msg_.transform,
+          world_base_link_trans);
+
+        tf2::Transform base_link_odom_trans;
+        bool can_transform_odom = ros_filter_utilities::lookupTransformSafe(
+          tf_buffer_.get(),
+          base_link_frame_id_,
+          odom_frame_id_,
+          filtered_position->header.stamp,
+          tf_timeout_,
```

Comment:
While I welcome the change, I think this behaviour is going to have to be a parameter, and it should default to `false` so we don't break anyone else's application. If we have to wait for the `odom`->`base_link` transform to be available, it could affect the latency and publication rate of the output (which is why we're using the latest available transform to begin with).

Reply:
Yes, that seems valid!

Reply:
Maybe `transform_timeout_odom_bl`?